### PR TITLE
ci: Add glint fix-your-editor message

### LINKED ISSUE #17008: Add message to fix editor on glinter format errors
### description

currently if glint fails its not easy to tell what the problem/fix is if you dont know in advance

it would be helpful to add general fix-your-editor advice in this case

### PR DESCRIPTION
**Pull request by phlax**
Commit Message: ci: Add glint fix-your-editor message
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue] Fix #17008 
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Comment by phlax** [comment]
with this PR the output on glint fail is:

```
TESTS FAILED:
  main@ /source/ci/format_pre.sh :45 (glint)
      Please fix your editor to ensure:
        - no trailing whitespace
        - no mixed tabs/spaces
        - all files end with a newline
##[error]Bash exited with code '1'.
Finishing: Run format pre-checks
```

File: tools/code_format/glint.sh [hunk 1]
```diff
@@ -20,7 +20,7 @@ check_mixed_tabs_spaces () {
     tabbed=$(grep -cP "^\t" "$1")
     spaced=$(grep -cP "^ " "$1")
     if [[ $tabbed -gt 0 ]] && [[ $spaced -gt 0 ]]; then
-        echo "mixed tabs and spaces: ${1}" >&2
+        echo "ERROR: mixed tabs and spaces: ${1}" >&2
```
Comment (jmarantz) [inline tools/code_format/glint.sh:23]: Does this syntax make the CI error window paint the error in red so it stands out?

The only line I see painted in red in an existing CI display is:
```
 ##[error]Bash exited with code '1'.
```
Maybe use that syntax?

See https://dev.azure.com/cncf/envoy/_build/results?buildId=79229&view=logs&j=c5dd2866-6ab3-5f3c-3a44-4cef0ec909b5&t=a9eb66d6-8944-5769-b3f7-476949dadcb8

Comment (phlax) [reply tools/code_format/glint.sh:23]: ill give it a whirl - it might be that console color codes are required if not - which is never pretty

Comment (phlax) [reply tools/code_format/glint.sh]: wow! that automagically worked, very handy!

**Comment by phlax** [comment]
`## [error]: ` added to final failure message also

should be ready for final review

File: ci/format_pre.sh [hunk 2]
```diff
@@ -50,9 +57,9 @@ CURRENT=extensions
 bazel run "${BAZEL_BUILD_OPTIONS[@]}" //tools/extensions:extensions_check
 
 if [[ "${#FAILED[@]}" -ne "0" ]]; then
-    echo "TESTS FAILED:" >&2
+    echo "##[error]: TESTS FAILED:" >&2
```
Comment (jmarantz) [inline ci/format_pre.sh:60]: looks good. Suggest making a bash variable ERROR_PREFIX and using it here and below. When you set BASH_PREFIX you can document why the syntax is desired.

Comment (phlax) [reply ci/format_pre.sh:60]: sure - ill update...

File: ci/format_pre.sh [hunk 3]
```diff
@@ -7,6 +7,7 @@
 
 FAILED=()
 CURRENT=""
+BASH_ERR_PREFIX="##[error]: "
```
Comment (jmarantz) [inline ci/format_pre.sh:10]: comment why this syntax?

The reason is that someday maybe we'll switch CI mechanisms (again) and want a different prefix for whatever the new system is, but won't know whether this is changeable.

**Review by jmarantz — APPROVED** [review]
Thanks!

File: ci/format_pre.sh (hunk 3 above)
Comment (phlax) [reply ci/format_pre.sh:11]: apologies - added

**Pull request merged**